lib/model/query/analytics.js

```diff
@@ -477,6 +475,84 @@ WHERE audits.action = 'entity.bulk.create'
 GROUP BY ds.id, ds."projectId"
 `);
 
+
+// Offline entities
+
+// Number of offline branches involving more than one update
+// Updates from offline-enabled Collect will include branchId so it is not enough
+// to count that but we can look at trunkVersion and branchBaseVersion to find
+// versions that had a true (multi-step) offlne operation.
+const countOfflineBranches = () => ({ oneFirst }) => oneFirst(sql`
+SELECT COUNT(DISTINCT "branchId")
+FROM entity_defs 
+WHERE "branchId" IS NOT NULL AND ("trunkVersion" IS NULL OR "branchBaseVersion" > "trunkVersion")
+`);
+
+// Look up offline branches that have another branchId
+// interrupting them, E.g. abc, abc, xyz, abc
+const countInterruptedBranches = () => ({ oneFirst }) => oneFirst(sql`
+WITH sortedRows AS (
+    SELECT 
+        "entityId",
+        "version",
+        "branchId",
+        LAG("branchId") OVER (PARTITION BY "entityId" ORDER BY "version") AS "prevBranchId"
+    FROM entity_defs
+),
+distinctRuns AS (
+    SELECT 
+        "entityId",
+        "branchId"
+    FROM sortedRows
+    WHERE "version" = 1 OR "branchId" IS DISTINCT FROM "prevBranchId" -- Keep first row and changes
+),
+duplicateRuns AS (
+    SELECT 
+        "entityId",
+        "branchId",
+        COUNT(*) AS runCount
+    FROM distinctRuns
+    WHERE "branchId" IS NOT NULL
+    GROUP BY "entityId", "branchId"
+    HAVING COUNT(*) > 1 -- Selects branchIds that occur more than once
+)
+SELECT COUNT(*)
+FROM duplicateRuns;
+`);
+
+// Number of submissions temporarily held in backlog but were automatically
+// removed from backlog when preceeding submission came in
+const countSubmissionReprocess = () => ({ oneFirst }) => oneFirst(sql`
+  SELECT COUNT(*)
+  FROM audits
+  WHERE "action" = 'submission.reprocess'
+`);
+
+// Measure how much time entities whose source is a submission.create
+// event take to process to look for a processing lag. We look at the
+// submission create loggedAt timestamp (when sub was created) to when
+// the event was processed, which will be after the entity version was
+// created.
+const measureEntityProcessingTime = () => ({ one }) => one(sql`
+SELECT
+  MAX("processed"-"loggedAt") as max_wait,
+  AVG("processed"-"loggedAt") as avg_wait
+FROM entity_def_sources
+JOIN audits ON audits.id = "auditId"
+WHERE action = 'submission.create'
+`);
+
+const measureElapsedEntityTime = () => ({ one }) => one(sql`
```

Comment:
I added this query for fun but decided not to include it in the actual report. Partly because it feels slower than some of the other queries and the overall set of queries is already sooo slow.

Reply:
Yeah, and I think as we discussed above, we would expect these metrics to be larger than those from `measureEntityProcessingTime()`, as these metrics will include the 5-day entity backlog wait and also the time between submission creation and submission approval (if submission approval is required).